src/auth/auth.service.ts

```
@@ -0,0 +1,305 @@
+    private userModel: Model<User>,
+    private jwtService: JwtService,
+    private emailService: EmailService, 
+  ) {}
+
+  async signUp(signUpDto: SignUpDto): Promise<void> {
+    const { firstName, lastName, email, password, role } = signUpDto;
+
+    const hashedPassword = await bcrypt.hash(password, 10);
+    const verificationToken = uuidv4();
+
+    const user = await this.userModel.create({
```

ESLint

[error] 'user' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)